commands/jobs.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Get Jobs from DevCongress Jobs board
+ * @author David Oddoye <oddoyedavid@gmail.com>
+ */
+'use strict';
+
+const request = require('request');
+const config = require('../config');
+
+const stopWords = [
+  'about', 'after', 'all', 'also', 'am', 'an', 'and', 'another', 'any', 'are', 'as', 'at', 'be',
+  'because', 'been', 'before', 'being', 'between', 'both', 'but', 'by', 'came', 'can',
+  'come', 'could', 'did', 'do', 'each', 'for', 'from', 'get', 'got', 'has', 'had',
+  'he', 'have', 'her', 'here', 'him', 'himself', 'his', 'how', 'if', 'in', 'into',
+  'is', 'it', 'like', 'make', 'many', 'me', 'might', 'more', 'most', 'much', 'must',
+  'my', 'never', 'now', 'of', 'on', 'only', 'or', 'other', 'our', 'out', 'over',
+  'said', 'same', 'see', 'should', 'since', 'some', 'still', 'such', 'take', 'than',
+  'that', 'the', 'their', 'them', 'then', 'there', 'these', 'they', 'this', 'those',
+  'through', 'to', 'too', 'under', 'up', 'very', 'was', 'way', 'we', 'well', 'were',
+  'what', 'where', 'which', 'while', 'who', 'with', 'would', 'you', 'your', 'a', 'i'];
+
+/**
+ * A method to get jobs from the DevCongress Jobs board API
+ * @param  String searchTerm
+ * @return JSON
+ */
+function getJobs(searchTerm) {
+  return new Promise((resolve, reject) => {
+    request(`${config.JOBS_URL}/?q=${searchTerm}`, {
+      headers: {
+        'Accept': 'application/json'
+      }
+    }, (error, response) => {
+      if (!error && response.statusCode == 200) {
+        const data = response.body;
+        resolve(JSON.parse(data));
+      } else {
+        // reject if an error, false for if there was no error but status code wasn't 200
+        reject(error || false);
+      }
+    });
+  });
+}
+
+module.exports = (bot, message) => {
+  bot.startTyping(message);
+  // Remove matched words (job, jobs) and stop words (https://en.wikipedia.org/wiki/Stop_words)
+  const searchTerms = message.text.split(' ').filter((word) => !stopWords.concat(['job', 'jobs']).includes(word));
+
+  // Make requests for all found words
+  const promises = searchTerms.map((searchTerm) => getJobs(searchTerm));
+  Promise.all(promises).then((arrayOfArrayOfJobs) => {
+    const jobsArray = arrayOfArrayOfJobs.reduce((result, arrayOfJobs) => result.concat(arrayOfJobs), []);
+    // remove duplicate jobs
+    const jobsAdded = new Set();
+    const uniqueJobs = [];
+
+    jobsArray.forEach((job) => {
+      if (!jobsAdded.has(job.url)) {
```

Comment:
So the use case here is the user types in `UI and UX jobs`. The handler strips the stop words and command, leaving us with _UI_ and _UX_. Requests to the API for bot words would return the same job currently, so i use the url as a means to determine unique jobs so we don't return duplicates

Reply:
@oddoye-david if I get this right, do you split the query and fire different requests for different words? if so then you don't have to. the search on the backend is trivial but non-trivial, as in, it can remove stop words and make the query generic enough so that it matches the most important cases. if you tried it and weren't satisfied with the results it produced then that's fine. but I'd be surprised if it couldn't "clean up" queries before running against our records.

Reply:
@oddoye-david I think this is the only comment holding this PR back from being merged, imo. I wanted to know if I got the gist of how the bot command works + suggest how to use the search utility provided the API without a lot of work.

Reply:
Cool cool. Updating now. Must have missed the comment before this one. All that christmas chicken

Reply:
Cool cool. Updating now. Must have missed the comment before this one. All that christmas chicken

Reply:
Thanks